query-engine/query-builders/query-builder/src/lib.rs

```diff
@@ -140,13 +141,54 @@ impl fmt::Display for RelationLink {
 }
 
 #[derive(Debug, Serialize)]
-pub struct DbQuery {
-    pub query: String,
-    pub params: Vec<PrismaValue>,
+#[serde(tag = "type", rename_all = "camelCase")]
+pub enum DbQuery {
+    RawSql {
+        sql: String,
+        params: Vec<PrismaValue>,
+    },
+    TemplateSql {
+        fragments: Vec<Fragment>,
+        placeholder: PlaceholderFormat,
+        params: Vec<PrismaValue>,
+    },
 }
 
 impl DbQuery {
-    pub fn new(query: String, params: Vec<PrismaValue>) -> Self {
-        Self { query, params }
+    pub fn get_params(&self) -> &Vec<PrismaValue> {
```

Comment:
```suggestion
    pub fn params(&self) -> &Vec<PrismaValue> {
```

See https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter